base/broadcast.jl

```diff
@@ -901,7 +888,8 @@ copy(bc::Broadcasted{<:Union{Nothing,Unknown}}) =
 const NonleafHandlingStyles = Union{DefaultArrayStyle,ArrayConflict}
 
 @inline function copy(bc::Broadcasted{Style}) where {Style}
-    ElType = combine_eltypes(bc.f, bc.args)
+    ElType = promote_typejoin_union(Base._return_type(_broadcast_getindex,
+                                                      Tuple{typeof(bc), Int}))
```

Comment:
I think this needs to be:

```suggestion
                                                      Tuple{typeof(bc), ndims(bc) == 1 ? typeof(axes(bc)[1]) : CartesianIndex{ndims(bc)}})
```

Reply:
Or is it `Base._return_type(iterate, Base._return_type(eachindex, Tuple{typeof(bc)}))` ?

Reply:
Oops, dropped a function. I meant:
```julia
index_type(bc) = iterate(eachindex(bc))[1]
Base._return_type(index_type, Tuple{typeof(bc)})
```

Reply:
Putting that all together:
```suggestion
_broadcast_getindex_eltype(bc) = _broadcast_getindex(bc, iterate(eachindex(bc))[1])
ElType = promote_typejoin_union(Base._return_type(_broadcast_getindex_eltype, Tuple{typeof(bc)}))
 ```

Reply:
I'm not entirely sure this is better than the existing code, which does pretty much the same calls, but bases it on calling `eltype`, instead of inference, which has at least _different_ tradeoffs for better or worse 🤔

Reply:
Do we want to try to proceed with this PR / design, or keep the current one?